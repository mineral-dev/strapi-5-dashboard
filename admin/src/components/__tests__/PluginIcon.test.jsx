import React from 'react';
import { render } from '@testing-library/react';
import { PluginIcon } from '../PluginIcon';

// Mock the Strapi icons
jest.mock('@strapi/icons', () => ({
  Coffee: () => <div data-testid="coffee-icon">Coffee Icon</div>,
}));

describe('PluginIcon', () => {
  it('should render without crashing', () => {
    const { container } = render(<PluginIcon />);
    expect(container).toBeTruthy();
  });

  it('should render the Coffee icon', () => {
    const { getByTestId } = render(<PluginIcon />);
    const coffeeIcon = getByTestId('coffee-icon');
    expect(coffeeIcon).toBeInTheDocument();
    expect(coffeeIcon).toHaveTextContent('Coffee Icon');
  });

  it('should match snapshot', () => {
    const { container } = render(<PluginIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });
});