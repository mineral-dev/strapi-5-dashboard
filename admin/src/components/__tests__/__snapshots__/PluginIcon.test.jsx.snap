// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginIcon should match snapshot 1`] = `
<div
  data-testid="coffee-icon"
>
  Coffee Icon
</div>
`;
